### PR TITLE
[new release] trace (2 packages) (0.3)

### PR DESCRIPTION
**Pull request by c-cube**
A stub for tracing/observability, agnostic in how data is collected

- Project page: <a href="https://github.com/c-cube/ocaml-trace">https://github.com/c-cube/ocaml-trace</a>

##### CHANGES:

- add explicit spans, for more precise tracing
- rename repo to ocaml-trace
- trace-tef: add a ticker thread to ensure we flush the file regularly

### CODE REVIEW

**Comment by c-cube** [comment]
There also needs to be a new upper bound for opentelemetry 0.5 (needs `trace < 0.3`)

File: packages/opentelemetry/opentelemetry.0.5/opam [hunk 1]
```diff
@@ -14,7 +14,7 @@ depends: [
   "pbrt" {>= "2.3"}
 ]
 depopts: [
-  "trace" {>= "0.1"}
+  "trace" {>= "0.1" & < "0.3"}
 ]
```
Comment (kit-ty-kate) [inline packages/opentelemetry/opentelemetry.0.5/opam:18]: See note in https://opam.ocaml.org/doc/Manual.html#opamfield-depopts
```suggestion
  "trace"
]
conflicts: [
  "trace" {< "0.1" | >= "0.3"}
]
```

**Comment by kit-ty-kate** [comment]
`tracy-client.0.1` seem to require an upper-bound constraint
```
#=== ERROR while compiling tracy-client.0.1 ===================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.14.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/tracy-client.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p tracy-client -j 71 @install
# exit-code            1
# env-file             ~/.opam/log/tracy-client-7-cc812e.env
# output-file          ~/.opam/log/tracy-client-7-cc812e.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I src/trace/.tracy_client_trace.objs/byte -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/trace -I /home/opam/.opam/4.14/lib/trace/core -I src/bindings/.tracy_client.objs/byte -intf-suffix .ml -no-alias-deps -o src/trace/.tracy_client_trace.objs/byte/tracy_client_trace.cmo -c -impl src/trace/tracy_client_trace.ml)
# File "src/trace/tracy_client_trace.ml", lines 5-30, characters 34-3:
#  5 | ..................................struct
#  6 |   let name_thread = Tracy_client.name_thread
#  7 |   let name_process _ = ()
#  8 | 
#  9 |   let process_data sp (d : string * Trace.user_data) =
# ...
# 27 |   let counter_float name n : unit = Tracy_client.plot name n
# 28 |   let counter_int name n : unit = counter_float name (float_of_int n)
# 29 |   let shutdown () = ()
# 30 | end
# Error: Signature mismatch:
#        ...
#        The value `with_span' is required but not provided
#        File "src/core/collector.ml", lines 20-27, characters 2-6:
#          Expected declaration
#        The value `enter_manual_span' is required but not provided
#        File "src/core/collector.ml", lines 34-42, characters 2-17:
#          Expected declaration
#        The value `exit_manual_span' is required but not provided
#        File "src/core/collector.ml", line 52, characters 2-46:
#          Expected declaration
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I src/trace/.tracy_client_trace.objs/byte -I src/trace/.tracy_client_trace.objs/native -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/trace -I /home/opam/.opam/4.14/lib/trace/core -I src/bindings/.tracy_client.objs/byte -I src/bindings/.tracy_client.objs/native -intf-suffix .ml -no-alias-deps -o src/trace/.tracy_client_trace.objs/native/tracy_client_trace.cmx -c -impl src/trace/tracy_client_trace.ml)
# File "src/trace/tracy_client_trace.ml", lines 5-30, characters 34-3:
#  5 | ..................................struct
#  6 |   let name_thread = Tracy_client.name_thread
#  7 |   let name_process _ = ()
#  8 | 
#  9 |   let process_data sp (d : string * Trace.user_data) =
# ...
# 27 |   let counter_float name n : unit = Tracy_client.plot name n
# 28 |   let counter_int name n : unit = counter_float name (float_of_int n)
# 29 |   let shutdown () = ()
# 30 | end
# Error: Signature mismatch:
#        ...
#        The value `with_span' is required but not provided
#        File "src/core/collector.ml", lines 20-27, characters 2-6:
#          Expected declaration
#        The value `enter_manual_span' is required but not provided
#        File "src/core/collector.ml", lines 34-42, characters 2-17:
#          Expected declaration
#        The value `exit_manual_span' is required but not provided
#        File "src/core/collector.ml", line 52, characters 2-46:
#          Expected declaration
```

**Review by kit-ty-kate — APPROVED** [review]

**Comment by mseri** [comment]
Thanks

**Pull request merged**